localstack/testing/pytest/fixtures.py

```diff
@@ -1940,7 +1940,15 @@ def create_rest_apigw(aws_client_factory):
 
     def _create_apigateway_function(**kwargs):
         region_name = kwargs.pop("region_name", None)
-        apigateway_client = aws_client_factory(region_name=region_name).apigateway
+        client_config = None
+        if is_aws_cloud():
+            client_config = botocore.config.Config(
+                # Api gateway can throttle requests pretty heavily. Leading to potentially undeleted apis
+                retries={"max_attempts": 10, "mode": "adaptive"}
+            )
+        apigateway_client = aws_client_factory(
+            region_name=region_name, config=client_config
+        ).apigateway
```

Comment:
Ah, this is a great way to fix it. We also have some other fixtures which just try deleting forever or almost if getting a 429, the throttling of `DeleteRestApi` is really insane to me, you'd need to add 15+ seconds sleep in between calls 😭 neat 👌